nixos/modules/system/boot/stage-2.nix

```diff
@@ -38,6 +38,16 @@ let
 in
 
 {
+  imports = [
+    (lib.mkRemovedOptionModule
+      [
+        "boot"
+        "readOnlyNixStore"
+      ]
+      "Please use the new option `boot.nixStoreMountOpts' to define mount options for the Nix store, including 'ro'"
```

Comment:
small point: Don't include the word `new`. It won't be new forever but this message will probably exist forever.